src/DynamoDbClientService.php

```diff
@@ -22,19 +23,33 @@ class DynamoDbClientService implements DynamoDbClientInterface
      */
     protected $attributeFilter;
 
-    public function __construct($config, Marshaler $marshaler, EmptyAttributeFilter $filter)
+    public function __construct(Marshaler $marshaler, EmptyAttributeFilter $filter)
     {
-        $this->client = new DynamoDbClient($config);
         $this->marshaler = $marshaler;
         $this->attributeFilter = $filter;
+        $this->clients = [];
     }
 
     /**
      * @return \Aws\DynamoDb\DynamoDbClient
      */
-    public function getClient()
+    public function getClient($connection = null)
     {
-        return $this->client;
+        $connection = $connection ?: config('dynamodb.default');
+
+        if (isset($this->clients[$connection])) {
+            return $this->clients[$connection];
+        }
+
+        $config = config("dynamodb.connections.$connection", []);
+        $config['version'] = '2012-08-10';
```

Comment:
Should the version be hard-coded here ... or should we allow the user to set it in the config, and only set it here if the user omits it?

Reply:
I thought about it too 🤔  but how our library is implemented quite depends on the version of the AWS API. This is like indicating our dependency.

Reply:
Ok, I can see that as a valid reason.